src/snowflake/cli/plugins/object/stage/manager.py

```diff
@@ -86,6 +85,32 @@ def get(
             f"get {self.quote_stage_name(stage_path)} {self._to_uri(dest_directory)} parallel={parallel}"
         )
 
+    def get_recursive(
+        self, stage_path: str, dest_path: Path, parallel: int = 4
+    ) -> List[SnowflakeCursor]:
+        list_files_result = self.list_files(stage_path).fetchall()
+        files_on_stage = [f["name"] for f in list_files_result]
```

Comment:
Should we consider extracting this to a method? Possibly an iterator
```py
def iter_stage(stage_path: str):
    for file in self.list_files(stage_path).fetchall():
        yield file['name']
```

In this way we will have a method that can be used in execute immediate

Reply:
Good idea

Reply:
Changed